staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -282,12 +286,11 @@ func ParseWatchCacheSizes(cacheSizes []string) (map[schema.GroupResource]int, er
 		if err != nil {
 			return nil, fmt.Errorf("invalid size of watch cache size: %s", c)
 		}
-		if size < 0 {
-			return nil, fmt.Errorf("watch cache size cannot be negative: %s", c)
-		}
-
 		watchCacheSizes[schema.ParseGroupResource(tokens[0])] = size
 	}
+	if _, ok := watchCacheSizes[schema.ParseGroupResource("events")]; !ok {
```

Comment:
This disallowed here for events @liggitt

Reply:
hard-coding a particular resource into k8s.io/apiserver isn't appropriate

Reply:
I agree on this, but we either have to store this information (If a resource needs a decorated storage or not) somewhere or provide it at start-up. I'm also not really a fan of adding a new flag for this.

Reply:
the map in NewHeuristicWatchCacheSizes seems like a better place to do something like 
`watchCacheSizes[schema.GroupResource{Resource: "events"}] = -1`

Reply:
sgtm